Cesium3DTilesSelection/src/TilesetLoadTilesetDotJson.cpp

```diff
@@ -430,9 +429,124 @@ void Tileset::LoadTilesetDotJson::Private::workerThreadLoadTerrainTile(
     }
   }
 
+  const auto availabilityLevelsIt =
+      layerJson.FindMember("metadataAvailability");
+
+  if (availabilityLevelsIt != layerJson.MemberEnd() &&
+      availabilityLevelsIt->value.IsInt()) {
+    context.availabilityLevels =
+        std::make_optional<uint32_t>(availabilityLevelsIt->value.GetInt());
+  } else {
+    std::vector<CesiumGeometry::QuadtreeTileRectangularRange>
+        availableTileRectangles =
+            QuantizedMeshContent::loadAvailabilityRectangles(layerJson, 0);
+    if (availableTileRectangles.size() > 0) {
+      for (const auto& rectangle : availableTileRectangles) {
+        context.implicitContext->rectangleAvailability->addAvailableTileRange(
+            rectangle);
+      }
+    }
+  }
+
+  const auto attributionIt = layerJson.FindMember("attribution");
+
+  if (attributionIt != layerJson.MemberEnd() &&
+      attributionIt->value.IsString()) {
+    context.credit = std::make_optional<Credit>(
+        context.pTileset->getExternals().pCreditSystem->createCredit(
+            attributionIt->value.GetString(),
+            context.pTileset->_options.showCreditsOnScreen));
+  }
+
+  std::string parentUrl =
+      JsonHelpers::getStringOrDefault(layerJson, "parentUrl", "");
+
+  if (!parentUrl.empty()) {
+    std::string resolvedUrl = Uri::resolve(context.baseUrl, parentUrl);
+    // append forward slash if necessary
+    if (resolvedUrl[resolvedUrl.size() - 1] != '/') {
+      resolvedUrl += '/';
+    }
+    resolvedUrl += "layer.json";
+    auto pAsyncSystem = context.pTileset->getAsyncSystem();
+    auto pAssetAccessor = context.pTileset->getExternals().pAssetAccessor;
+    pAssetAccessor->get(pAsyncSystem, resolvedUrl, context.requestHeaders)
```

Comment:
This future should be returned here, too, so that the outer future doesn't resolve into the full chain of parent URL loading futures resolves.

Reply:
In the case where the parentUrl is empty, just return `asyncSystem.createResolvedFuture()`.